### ISSUE TITLE
Rule: use only arrow functions in React component 

### ISSUE THREAD

**Issue by spanwair** [yannickcr/eslint-plugin-react#2723]
Eslint v: v7.5.0

### **Description**

I got class defining inisideFunc(). I get this code, exmp:

```
export class Hello {
  insideFunc() { ... }
  render() { ... }
}
```

but I require to use only arrow functions instead normal function, except render, exmp:

```
export class Hello {
  insideFunc = () => { ... }
  render() { ... }
}
```

### **What I did**
I already tried func-style: "error" or third party library like eslint-plugin-prefer-arrow but it didn't show any error, exmp:

```
func-style: "error"

    "prefer-arrow/prefer-arrow-functions": [
     "error",
     {
       "disallowPrototype": true,
       "singleReturnOnly": false,
       "classPropertiesAllowed": false
     }
   ]
```

### **What I expect**

I would like to use only arrow functions in my React component

**Comment by spanwair** [comment]
I do not think is invalid repo because this issue is only about React eslint validation where all rules fail. Mainly because in component we do not have functions but methods and that why we can't use any solution. In our app we use only arrow functions and this was the rule from the beginning. In TS it would be `force-arrow-method` but it do not has yet any equivalent in eslint

**Comment by ljharb** [comment]
I see; you're asking for that rule, and eslint-plugin-arrow was just something you tried that didn't work.

In that case, I'd still decline to support that rule - "only use arrow functions" is just a flat-out incorrect way to use the language, and it would be very harmful imo to make it easier for people to opt into that coding style.

**Comment by spanwair** [comment]
I can't decide to make it different App in continuously developed for over 3 years and before that someone decided to use this code style. I am just migrating it from TSlint to ESlint and only this rule make it difficult for me.

**Comment by ljharb** [comment]
Performance is only one of the concerns, and the least important one. It's a philosophy/clarity concern.

I'd encourage you to write your own eslint plugin for this rule.

**Comment by spanwair** [comment]
can you point more of the concerns or forward me to some article about it?

**Comment by ljharb** [comment]
https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1
